lib/menubar/utils.lua

```diff
@@ -214,29 +214,28 @@ function utils.lookup_icon_uncached(icon_file)
         return false
     end
 
-    local icon_file_ext = icon_file:match(".*%.(.-)$")
-    if icon_file:sub(1, 1) == '/' and supported_icon_formats[icon_file_ext] then
-        -- If the path to the icon is absolute and its format is
-        -- supported, do not perform a lookup.
+    local icon_file_ext = icon_file:match(".+%.(.*)$")
+    if icon_file:sub(1, 1) == '/' and supported_icon_file_exts[icon_file_ext] then
```

Comment:
Above is the regex change - The rest are pretty much renaming variables to make them clearer and editing the comments